gensim/downloader.py

```diff
@@ -0,0 +1,298 @@
+                .format(base_dir))
+
+
+def initialize_data_log_file():
+    """Function for initializing the log file. Creates a json object
+    for each corpus/model and stores in the log file. For eg: {"name": "text8", "status" : "None"}
+    """
+    data = info()
+    corpora = data['gensim']['corpus']
+    models = data['gensim']['model']
+    json_list = []
+    for corpus in corpora:
+        json_object = {"name": corpus, "status": "None"}
+        json_list.append(json_object)
+    for model in models:
+        json_object = {"name": model, "status": "None"}
+        json_list.append(json_object)
+    json.dump(json_list, data_log_file)
+    data_log_file.close()
+
+
+def update_data_log_file(dataset, status):
+    """Function for updating the status of the dataset json object.
+
+    Args:
+        dataset(string): Name of the corpus/model.
+        status(string): Status to be updates to i.e downloaded or installed.
+    """
+    jdata = json.loads(open(data_log_file_dir).read())
+    for json_object in jdata:
+        if json_object["name"] == dataset:
+            json_object["status"] = status
+    with open(data_log_file_dir, 'w') as f:
+        f.write(json.dumps(jdata))
+
+
+def get_data_status(dataset):
+    """Function for finding the status of the dataset.
+
+    Args:
+        dataset(string): Name of the corpus/model.
+
+    Returns:
+        string: returns the current status of the corpus/model i.e None, downloaded or installed.
+    """
+    jdata = json.loads(open(data_log_file_dir).read())
+    for json_object in jdata:
+        if json_object["name"] == dataset:
+            return json_object["status"]
+
+
+def calculate_md5_checksum(folder_dir):
+    """Function for calculating checksum of a downloaded model/corpus.
+
+    Args:
+        folder_dir(string): Path to the downloaded model.
+
+    Returns:
+        string: It returns the value for the checksum for folder_dir directory
+    """
+    hash_md5 = hashlib.md5()
+    for filename in os.listdir(folder_dir):
+        file_dir = os.path.join(folder_dir, filename)
+        with open(file_dir, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+    return hash_md5.hexdigest()
+
+
+def info(dataset=None):
+    """Function for retrieving the list of corpora/models, if dataset is not provided. If dataset
+    is provided, then it gives detailed information about the dataset.
+
+    Args:
+        dataset(string): Name of the corpus/model.
+
+    Returns:
+        <class 'dict'>: It returns the models/corpora names with detailed information about each.
+    """
+    url = "https://raw.githubusercontent.com/chaitaliSaini/Corpus_and_models/master/list_with_filename.json"
+    response = urlopen(url)
+    data = response.read().decode("utf-8")
+    data = json.loads(data)
+    if dataset is not None:
+        corpora = data['gensim']['corpus']
```

Comment:
`gensim` is useless key in your json, you need only `model` and `dataset` keys for upper level